### DISCUSSION TITLE
Multiple Categories for a test case

### DISCUSSION THREAD

**Discussion by danpost321** [xunit/xunit#2197] [Question] [answered]
How do I assign multiple Categories to test cases?

For example, I want to be able to run all the tests when Category="Regression", I only want FirstTest and ThirdTest when Category = "Smoke", etc.

Example:

```csharp
[Trait("Category", "Smoke")] 
[Trait("Category", "Regression")]
private FirstTest()
{ 
    /* some test code */
}

[Trait("Category", "Regression")]
private SecondTest()
{ 
    /* some test code */
}

[Trait("Category", "Smoke")] 
[Trait("Category", "Regression")]
[Trait("Category", "FoundByCEO")]
private ThirdTest()
{ 
    /* some test code */
}
```

**Comment by bradwilson** [comment, marked as answer]
Actually, you do it exactly like you did. :) Each trait name (in this case, `Category`) is allowed to have multiple values, and the simplest way to get multiple values is the way you did it with the `[Trait]` attribute.

**Reply by bradwilson** [reply]
Going to mark this as answered. Please let me know if you have any further questions.
